src/pudl/output/ferc1.py

```diff
@@ -871,3 +873,529 @@ def add_mean_cap_additions(steam_df):
+            value_cols.append(
+                pudl.transform.ferc1.FERC1_TFR_CLASSES[
+                    table_name
+                ]().params.reconcile_table_calculations.column_to_check
+            )
+        if len(set(value_cols)) != 1:
+            raise ValueError(
+                "Exploding FERC tables requires tables with only one value column. Got: "
+                f"{set(value_cols)}"
+            )
+        value_col = list(set(value_cols))[0]
+        return value_col
+
+    def boom(
+        self,
+        tables_to_explode: dict[str, pd.DataFrame],
+        clean_xbrl_metadata_json: dict,
+    ) -> pd.DataFrame:
+        """Explode a set of nested tables.
+
+        There are five main stages of this process:
+
+        #. Prep all of the individual tables for explosion.
+        #. Concatenate all of the tabels together.
+        #. Remove duplication in the concatenated exploded table.
+        #. Add in metadata for the remaining ``xbrl_factoid`` s regarding linage (not
+           implemented yet).
+        #. Validate (not implemented yet).
+
+        Args:
+            tables_to_explode: dictionary of table name (key) to transfomed table (value).
+            root_table: the table at the base of the tree of tables_to_explode.
+            clean_xbrl_metadata_json: json version of the XBRL-metadata.
+        """
+        exploded = (
+            self.initial_explosion_concatenation(
+                tables_to_explode, clean_xbrl_metadata_json
+            )
+            # REMOVE THE DUPLICATION
+            .pipe(self.remove_factoids_from_mutliple_tables, tables_to_explode)
+            .pipe(self.remove_totals_from_other_dimensions)
+            .pipe(self.remove_inter_table_calc_duplication)
+            .pipe(remove_intra_table_calculated_values)
+        )
+        return exploded
+
+    def initial_explosion_concatenation(
+        self, tables_to_explode: dict[str, pd.DataFrame], clean_xbrl_metadata_json: dict
+    ) -> pd.DataFrame:
+        """Concatenate all of the tables for the explosion.
+
+        Merge in some basic pieces of the each table's metadata and add ``table_name``. At
+        this point in the explosion, there will be a lot of duplicaiton in the output.
+        """
+        logger.info("Explode: CONCAT!")
+        explosion_tables = []
+        # GRAB/PREP EACH TABLE
+        for table_name, table_df in tables_to_explode.items():
+            xbrl_factoid_name = pudl.transform.ferc1.FERC1_TFR_CLASSES[
+                table_name
+            ]().params.xbrl_factoid_name
+            tbl = table_df.assign(table_name=table_name).rename(
+                columns={xbrl_factoid_name: "xbrl_factoid"}
+            )
+            explosion_tables.append(tbl)
+        metadata_exploded = self.meta_exploder.boom(clean_xbrl_metadata_json)
+        exploded = pd.concat(explosion_tables)
+        # drop any metadata columns coming from the tbls bc we may have edited the
```

Comment:
@cmgosnell I think that doing this drops the metadata we manufactured for the corrected factoids, since it isn't in clean_xbrl_metadata_json.